### ISSUE TITLE
The databases lose the linking

### ISSUE THREAD

**Issue by dummifiedme** [connertennery/Notion-to-Obsidian-Converter#10]
The database after conversion is shown as a markdown table, but the "link" is gone. Is it something which is difficult to get done or maybe skipped while programmming?

I am very thankful to you for this code btw 👍 
Just noticed this and thought I should post.

**Comment by connertennery** [comment]
Thank you for creating this issue! Right now I don't think it attempts to recreate the links, but I'll try to add it in when I get the chance.

Glad it's able to help you out! 👍

**Comment by dummifiedme** [comment]
> Thank you for creating this issue! Right now I don't think it attempts to recreate the links, but I'll try to add it in when I get the chance.
> 
> Glad it's able to help you out! 👍

Hi,
Were you able to work on this? :)

**Comment by connertennery** [comment]
> Hi,
> Were you able to work on this? :)

@dummifiedme I haven't had the time due to work. I'm sorry I haven't been able to get around to it yet, but if you'd like to correct it before I do and submit a pull request I'll look it over.

**Comment by dummifiedme** [comment]
> > Hi,
> > Were you able to work on this? :)
> 
> @dummifiedme I haven't had the time due to work. I'm sorry I haven't been able to get around to it yet, but if you'd like to correct it before I do and submit a pull request I'll look it over.

I don't have such expertise in coding. Its fine, take your time. I am glad you have made the code :)

**Comment by connertennery** [comment]
@dummifiedme I just looked through the code and I realized it did try to create the necessary links so I'm a little confused why you're not seeing them. I tested it with my own Notion export and it created the links in the Markdown table and they're functional.

Would you mind providing your export to me? You could cut it down to just a database and a few of the pages it links to. If you do that, please keep the directory structure as exported by Notion. I understand if you aren't comfortable sharing that, so if you'd like, you could attempt to recreate your setup using dummy data and then export that. I'll test with that and see what's causing the issue.

Hope this helps!

**Comment by dummifiedme** [comment]
I will get back to you as soon as possible.
Till now, I tried multipl times across several days, and everytime I got markdown tables without may links to the page. 

I will share you the export. I will have to create dummy data, as my own notes are not well organized as of now. (still in the process of setting up my workflow).

**Comment by dummifiedme** [comment]
The Page without DB - Link to the page is there.
![image](https://user-images.githubusercontent.com/68785729/97170287-6e2c9d80-17b1-11eb-9657-3f2f4ac43b75.png)

The page with DB - Link to the pages A , B, C missing. (@pageC is just a typo, it was to be #pageC, doesn't matter in the current question)
![image](https://user-images.githubusercontent.com/68785729/97170236-52c19280-17b1-11eb-858b-80c6d3ba6383.png)

Here's an export
[Export-9ac2d544-e077-48e2-8c80-e4d50645a536.zip](https://github.com/connertennery/Notion-to-Obsidian-Converter/files/5438607/Export-9ac2d544-e077-48e2-8c80-e4d50645a536.zip)

**Comment by dummifiedme** [comment]
@connertennery were you able to have a look at it?

**Comment by connertennery** [comment]
@dummifiedme I took a look at it but the export itself doesn't have page links. Typically the exported pages Are referenced by their page, ending with `.md` in the cell of the table. I'm going to look a little more into it and see what I can figure out. Maybe it's how Notion exports this kind of database.

**Comment by connertennery** [comment]
@dummifiedme Okay so I think I know what the problem is. Each record in a database is a new page that gets exported. The first field is the ID and that's just plain text instead of being a link like in other places. When you use a relational column or a roll up column, it creates a `/Path/To/Page.md` link in the cell.

So my idea is that I can attempt to link the first cell, but any comma in the ID field could make that more difficult. Another route is that I could check the entire line and see if any of the nested page names are within the line then link it, but that could also be problematic.

When I get more time I'll try to work a bit on it and see if I can make it create links, but at best they'll be informed guesses. I should have some more time over the next few days so I'll try to address it soon. I hope this helps. :)

**Comment by dummifiedme** [comment]
> @dummifiedme Okay so I think I know what the problem is. Each record in a database is a new page that gets exported. The first field is the ID and that's just plain text instead of being a link like in other places. When you use a relational column or a roll up column, it creates a `/Path/To/Page.md` link in the cell.
> 
> So my idea is that I can attempt to link the first cell, but any comma in the ID field could make that more difficult. Another route is that I could check the entire line and see if any of the nested page names are within the line then link it, but that could also be problematic.
> 
> When I get more time I'll try to work a bit on it and see if I can make it create links, but at best they'll be informed guesses. I should have some more time over the next few days so I'll try to address it soon. I hope this helps. :)

I thought every new entry in Notion database is essentially a page by default. We could use a link to another page as well, but that will have me create a page somewhere else. 

From what I understand.

- Notion exports a DB as csv and a folder with MD files.
- we just need to take the csv file name, search for a folder with the same name in the same directory, all the files inside this folder would form a part of database. 
- notion to obsidian converter already removes those encryption or code like things from the names.
- now we just have to look for the names of the MD files from second point,in the csv table. And replace it with the link of the file.
- it seems workable?

**Comment by connertennery** [comment]
@dummifiedme Yeah, I ended up having a few minutes earlier so I wrote the code in. Pull latest and see if that works for you.

Every record is a page, but the exported CSV doesn't link the pages for the identity field like it does when you reference a page in other columns. I just look through the nested directory and check for a matching name then link it up. It worked with the provided example database, but that's a pretty small sample. Test it out and let me know if there are any more problems.
